src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -67,6 +73,93 @@
+                    awardExtension.getBudgetBeginningDate());
+            localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_BUDGET_END_DATE, awardExtension.getBudgetEndingDate());
+            localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_BUDGET_TOTAL, awardExtension.getBudgetTotalAmount());
+            localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_PRIME_AGREEMENT_NUMBER,
+                    awardExtension.getPrimeAgreementNumber());
+        }
+    }
+
+    protected void setPurchaseOrderNumberFieldInParameterMap(ContractsGrantsInvoiceDocument document, Map<String, Object> localParameterMap) {
+        Award award = (Award) document.getInvoiceGeneralDetail().getAward();
+        String invoiceOptionCode = award.getInvoicingOptionCode();
+        String purchaseOrderNumber = StringUtils.EMPTY;
+
+        if (StringUtils.equals(invoiceOptionCode, CGIntegrationConstants.AwardInvoicingOption.Types.AWARD.getCode())) {
+            purchaseOrderNumber = findPurchaseOrderNumberForInvoiceOptionAward(award);
+        } else if (StringUtils.equals(invoiceOptionCode, CGIntegrationConstants.AwardInvoicingOption.Types.ACCOUNT.getCode())) {
+            purchaseOrderNumber = findPurchaseOrderNumberForInvoiceOptionAccoiunt(document);
+        } else if (StringUtils.equals(invoiceOptionCode, CGIntegrationConstants.AwardInvoicingOption.Types.CONTRACT_CONTROL.getCode())
+                || StringUtils.equals(invoiceOptionCode, CGIntegrationConstants.AwardInvoicingOption.Types.SCHEDULE.getCode())) {
+            purchaseOrderNumber = findPurchaseOrderNymberForInvoiceOptionContractControlAndScheduled(document);
+        }
+        
+        if (StringUtils.isBlank(purchaseOrderNumber)) {
+            LOG.error("setPurchaseOrderNumberFieldInParameterMap, unable to find the purchase order number for document " + document.getDocumentNumber());
+        }
+        
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("setPurchaseOrderNumberFieldInParameterMap, for document " + document.getDocumentNumber() + " with an invoice option code of "
+                    + invoiceOptionCode + " the purchase order number is " + purchaseOrderNumber);
+        }
+
+        localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_PURCHASE_ORDER_NBR, purchaseOrderNumber);
+    }
+
+    protected String findPurchaseOrderNumberForInvoiceOptionAward(Award award) {
+        String purchaseOrderNumber = StringUtils.EMPTY;
+        AwardExtendedAttribute awardExtension = (AwardExtendedAttribute) award.getExtension();
+        if (ObjectUtils.isNotNull(awardExtension)) {
+            purchaseOrderNumber = awardExtension.getPurchaseOrderNumber();
+        }
+        return purchaseOrderNumber;
+    }
+
+    protected String findPurchaseOrderNumberForInvoiceOptionAccoiunt(ContractsGrantsInvoiceDocument document) {
```

Comment:
Minor typo in method name; should be changed from "...Accoiunt" to "...Account"

Reply:
ok